plugins/inara.py

```diff
@@ -1141,74 +1142,84 @@ def call(callback=None, force=False):
-        item = this.queue.get()
-        if not item:
-            return  # Closing
-        else:
-            (url, data, callback) = item
-
-        retrying = 0
-        while retrying < 3:
-            try:
-                r = this.session.post(url, data=json.dumps(data, separators=(',', ':')), timeout=_TIMEOUT)
-                r.raise_for_status()
-                reply = r.json()
-                status = reply['header']['eventStatus']
-                if callback:
-                    callback(reply)
-
-                elif status // 100 != 2:  # 2xx == OK (maybe with warnings)
-                    # Log fatal errors
-                    logger.warning(f'Inara\t{status} {reply["header"].get("eventStatusText", "")}')
-                    logger.debug(f'JSON data:\n{json.dumps(data, indent=2, separators = (",", ": "))}')
-                    plug.show_error(_('Error: Inara {MSG}').format(MSG=reply['header'].get('eventStatusText', status)))
+        to_send: Tuple[str, OrderedDictT[str, Any], None] = this.queue.get()
+        if to_send is None:
+            # Asked to exit
+            return
 
-                else:
-                    # Log individual errors and warnings
-                    for data_event, reply_event in zip(data['events'], reply['events']):
-                        if reply_event['eventStatus'] != 200:
-                            logger.warning(f'Inara\t{status} {reply_event.get("eventStatusText", "")}')
-                            logger.debug(f'JSON data:\n{json.dumps(data_event)}')
-                            if reply_event['eventStatus'] // 100 != 2:
-                                plug.show_error(_('Error: Inara {MSG}').format(
-                                    MSG=f'{data_event["eventName"]},'
-                                        f'{reply_event.get("eventStatusText", reply_event["eventStatus"])}'))
-
-                        if data_event['eventName'] in ('addCommanderTravelCarrierJump',
-                                                       'addCommanderTravelDock',
-                                                       'addCommanderTravelFSDJump',
-                                                       'setCommanderTravelLocation'):
-                            this.lastlocation = reply_event.get('eventData', {})
-                            # calls update_location in main thread
-                            this.system_link.event_generate('<<InaraLocation>>', when="tail")
-
-                        elif data_event['eventName'] in ['addCommanderShip', 'setCommanderShip']:
-                            this.lastship = reply_event.get('eventData', {})
-                            # calls update_ship in main thread
-                            this.system_link.event_generate('<<InaraShip>>', when="tail")
+        url, data, _ = to_send
+        try_send_data(url, data)
+        time.sleep(FLOOD_LIMIT_SECONDS)
 
+
+def try_send_data(url: str, data: OrderedDictT[str, Any]):
+    for i in range(3):
```

Comment:
if the below-mentioned return is added, this can log at the end of the loop that sending failed etc.